File: MaximumNumberOfPairsInArray/easyProblem.js
// You are given a 0-indexed integer array nums. In one operation, you may do the following:

// Choose two integers in nums that are equal.
// Remove both integers from nums, forming a pair.
// The operation is done on nums as many times as possible.

// Return a 0-indexed integer array answer of size 2 where answer[0] is the number of pairs that are formed and answer[1] is the number of leftover integers in nums after doing the operation as many times as possible.

/**
 * @param {number[]} nums
 * @return {number[]}
 */
var numberOfPairs = function (nums) {
  let count = 0;

  let x = nums.sort((a, b) => a - b);
  for (let i = 0; i < x.length; i++) {
    if (x[i] == x[i + 1]) {
      x.shift();
      x.shift();
      i--;
      count++;
    }
  }

  return [count, x.length];
};

File: N-RepeatedElementInSize2nArray/easyProblem.java
// You are given an integer array nums with the following properties:
// nums.length == 2 * n.
// nums contains n + 1 unique elements.
// Exactly one element of nums is repeated n times.
// Return the element that is repeated n times.



class Solution {
    public int repeatedNTimes(int[] A) {
        Map<Integer, Integer> count = new HashMap();
        for (int x: A) {
            count.put(x, count.getOrDefault(x,0) + 1);
        }
        
        for (int k: count.keySet())
            if (count.get(k) > 1)
                return k;

        throw null;
    }
}